DDDEastAnglia.Tests/InMemoryRepository.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace DDDEastAnglia.Tests
+{
+    internal class InMemoryRepository<T>
+    {
+        private readonly Func<T, int> keyFunc;
+        private readonly ConcurrentDictionary<int, T> store = new ConcurrentDictionary<int, T>();
+
+        public InMemoryRepository(Func<T, int> keyFunc)
+        {
+            this.keyFunc = keyFunc;
```

Comment:
Do the null checks actually help? We are using Ninject which will blow up when it can't create find a dependency.. so the null checks will never throw?